core/lib/via_btc_client/src/inscriber/mod.rs

```diff
@@ -1 +1,737 @@
+
+        for (outpoint, txout) in utxos {
+            let txin = TxIn {
+                previous_output: outpoint,
+                script_sig: ScriptBuf::default(), // For a p2wpkh script_sig is empty.
+                sequence: Sequence::ENABLE_RBF_NO_LOCKTIME,
+                witness: Witness::default(), // Get filled in after signing.
+            };
+
+            commit_tx_inputs.push(txin);
+            unlocked_value += txout.value;
+            inputs_count += 1;
+            utxo_amounts.push(txout.value);
+        }
+
+        let res = CommitTxInputRes {
+            commit_tx_inputs,
+            unlocked_value,
+            inputs_count,
+            utxo_amounts,
+        };
+
+        Ok(res)
+    }
+
+    fn is_p2wpkh(&self, script_pubkey: &ScriptBuf) -> bool {
+        let p2wpkh_script = self.signer.get_p2wpkh_script_pubkey();
+
+        script_pubkey == p2wpkh_script
+    }
+
+    async fn prepare_commit_tx_output(
+        &self,
+        tx_input_data: &CommitTxInputRes,
+        inscription_pubkey: ScriptBuf,
+    ) -> Result<CommitTxOutputRes> {
+        let inscription_commitment_output = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: inscription_pubkey,
+        };
+
+        let mut fee_rate = self.get_fee_rate().await?;
+        let pending_tx_in_context = self.context.fifo_queue.len();
+
+        // increase fee rate based on pending transactions in context
+
+        let increase_factor = FEE_RATE_INCREASE_PER_PENDING_TX * pending_tx_in_context as u64;
+        fee_rate += fee_rate * increase_factor / 100;
+
+        let fee_amount = InscriberFeeCalculator::estimate_fee(
+            tx_input_data.inputs_count,
+            COMMIT_TX_P2TR_INPUT_COUNT,
+            COMMIT_TX_P2WPKH_OUTPUT_COUNT,
+            COMMIT_TX_P2TR_OUTPUT_COUNT,
+            vec![],
+            fee_rate,
+        )?;
+
+        let commit_tx_change_output = TxOut {
+            value: tx_input_data.unlocked_value - fee_amount,
+            script_pubkey: self.signer.get_p2wpkh_script_pubkey().clone(),
+        };
+
+        let res = CommitTxOutputRes {
+            commit_tx_change_output,
+            commit_tx_tapscript_output: inscription_commitment_output,
+            commit_tx_fee_rate: fee_rate,
+            _commit_tx_fee: fee_amount,
+        };
+
+        Ok(res)
+    }
+
+    async fn get_fee_rate(&self) -> Result<u64> {
+        let res = self.client.get_fee_rate(FEE_RATE_CONF_TARGET).await?;
+
+        Ok(res)
+    }
+
+    fn sign_commit_tx(
+        &self,
+        input: &CommitTxInputRes,
+        output: &CommitTxOutputRes,
+    ) -> Result<FinalTx> {
+        let temp_tx_out1 = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: ScriptBuf::default(),
+        };
+        let temp_tx_out2 = TxOut {
+            value: Amount::ZERO,
+            script_pubkey: ScriptBuf::default(),
+        };
+
+        let mut commit_outputs: [TxOut; 2] = [temp_tx_out1, temp_tx_out2];
```

Comment:
Maybe something like this would be more elegant solution for initializing the TxOut array
```suggestion
        let mut commit_outputs: [TxOut; 2] = [TxOut::NULL, TxOut::NULL];
```